components/encoder/fake/encoder.go

```diff
@@ -109,27 +111,24 @@ func (e *fakeEncoder) Position(
 
 // Start starts a background thread to run the encoder.
 func (e *fakeEncoder) start(cancelCtx context.Context) {
-	e.activeBackgroundWorkers.Add(1)
-	utils.ManagedGo(func() {
-		for {
-			select {
-			case <-cancelCtx.Done():
-				return
-			default:
-			}
-
-			e.mu.RLock()
-			updateRate := e.updateRate
-			e.mu.RUnlock()
-			if !utils.SelectContextOrWait(cancelCtx, time.Duration(updateRate)*time.Millisecond) {
-				return
-			}
-
-			e.mu.Lock()
-			e.position += int64(e.speed / float64(60*1000/updateRate))
-			e.mu.Unlock()
+	for {
+		select {
+		case <-cancelCtx.Done():
+			return
+		default:
 		}
-	}, e.activeBackgroundWorkers.Done)
+
+		e.mu.RLock()
+		updateRate := e.updateRate
+		e.mu.RUnlock()
+		if !utils.SelectContextOrWait(cancelCtx, 100*time.Millisecond) {
+			return
+		}
+
+		e.mu.Lock()
+		e.position += int64(e.ticksPerSec / float64(1000/updateRate))
```

Comment:
This thing can never go backwards, and would need a dependency on motor to figure out if it would have to. Needs a bit of thought.

Reply:
this is the same behavior as `speed` from before though right? we can pass in a negative `ticksPerSec` as well. Additionally this is still a fake encoder, not intended to be used with a real motor iirc

Reply:
They used to be. Right now, we can do this since we needed the fake encoder from gpio motor's encoded_motor_tests; it used to be that the tests called `SetSpeed` directly use the encoder in tests. I think if we're going to add functionality, we should think a bit more about this, maybe migrate the SetSpeed and SetPosition to DoCommands to be cleaner? That we it's more representative of an actual encoder.